staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -130,6 +136,7 @@ func tlsConfigKey(c *Config) (tlsCacheKey, error) {
 		nextProtos:         strings.Join(c.TLS.NextProtos, ","),
 		dial:               fmt.Sprintf("%p", c.Dial),
 		disableCompression: c.DisableCompression,
+		proxier:            fmt.Sprintf("%p", c.Proxier),
```

Comment:
add tests to TestTLSConfigKey uniqueConfigurations that ensures Proxier: {nil, http.ProxyFromEnvironment, <custom func>} all produce distinct cache keys

Reply:
Done.